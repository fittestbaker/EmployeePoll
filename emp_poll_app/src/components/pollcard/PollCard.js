import "./PollCard.css";
import { formatDate } from "../../helpers/formatTimestamp";
import { Link } from "react-router-dom";
import { useNavigate } from "react-router-dom";
import { useState } from "react";

const PollCard = ({ question }) => {
  const navigate = useNavigate();
  const [submit, setSubmit] = useState(false);

  const handleOnClick = (e) => {
    e.preventDefault();
    setSubmit(true);
    const qid = e.target.value;
    navigate("/questions/" + qid);
  };

  return (
    <div id="questionCard">
      {submit ?? <p>Showing Question</p>}
      <p>{question.author}</p>
      <p>{formatDate(question.timestamp)}</p>
      <button onClick={handleOnClick} value={question.id}>
        Show
      </button>
    </div>
  );
};

export default PollCard;
