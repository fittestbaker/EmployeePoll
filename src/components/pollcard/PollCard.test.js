import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import PollCard from "./PollCard";
import { MemoryRouter } from "react-router";

describe("render PollCard in document", () => {
  it("should be successful", async () => {
    const questionInput = {
      author: "sarahedo",
      timestamp: 1467166872634,
      id: "6ni6ok3ym7mf1p33lnez",
    };
    render(
      <MemoryRouter>
        <PollCard question={{ questionInput }} />
      </MemoryRouter>
    );

    const button = screen.getByText(/show/i);
    expect(button).toBeInTheDocument();
  });
});

describe("render PollCard snapshot", () => {
  it("should be successful", async () => {
    const questionInput = {
      author: "sarahedo",
      timestamp: 1467166872634,
      id: "6ni6ok3ym7mf1p33lnez",
    };
    var { component } = render(
      <MemoryRouter>
        <PollCard question={{ questionInput }} />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
  });
});

describe("render PollCard click", () => {
  it("should be successful", async () => {
    const questionInput = {
      author: "sarahedo",
      timestamp: 1467166872634,
      id: "6ni6ok3ym7mf1p33lnez",
    };
    render(
      <MemoryRouter>
        <PollCard question={{ questionInput }} />
      </MemoryRouter>
    );

    const button = screen.getByText(/Show/i);
    fireEvent.click(button);

    setTimeout(() => {
      const pollElement = screen.getByText(/Showing/i);
      expect(pollElement).toBeInTheDocument();
    }, 1000);
  });
});
