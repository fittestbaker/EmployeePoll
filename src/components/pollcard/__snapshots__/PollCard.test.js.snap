// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PollCard snapshot should be successful 1`] = `undefined`;
